pset/updater.go

```diff
@@ -380,29 +386,34 @@ type AddIssuanceArg struct {
 }
 
 //AddIssuance adds non-confidential issuance to the transaction
-func (p *Updater) AddIssuance(arg AddIssuanceArg) (uint32, error) {
+func (p *Updater) AddIssuance(arg AddIssuanceArg) error {
 	if len(p.Upsbt.UnsignedTx.Inputs) == 0 {
-		return 0, errors.New("transaction must contain at least one input")
+		return errors.New("transaction must contain at least one input")
 	}
 
 	if arg.AssetAmount == 0 {
-		return 0, errors.New("asset amount must be greater then 0")
+		return errors.New("asset amount must be greater then 0")
 	}
 
 	if arg.AssetAddress == "" {
-		return 0, errors.New("destination address for issued asset must" +
+		return errors.New("destination address for issued asset must" +
 			" be provided")
 	}
 
 	if arg.TokenAmount > 0 && arg.TokenAddress == "" {
-		return 0, errors.New("destination address for reissuance token " +
+		return errors.New("destination address for reissuance token " +
 			"must be provided")
 	}
 
+	if arg.TokenFlag != NonConfidentialReissuanceTokenFlag &&
+		arg.TokenFlag != ConfidentialReissuanceTokenFlag {
+		return errors.New("token flag must be 0 or 1")
+	}
+
```

Comment:
maybe it is the case to perform these checks only when the token amount is defined? 
```suggestion
	if arg.TokenAmount > 0 {
	        if arg.TokenAddress == "" {
	            return errors.New("destination address for reissuance token " +
			"must be provided")
	        }
	        if arg.TokenFlag != NonConfidentialReissuanceTokenFlag &&
		    arg.TokenFlag != ConfidentialReissuanceTokenFlag {
		    return errors.New("token flag must be 0 or 1")
		}
	}
```
This way we would just ignore token params when its amount is not defined

Reply:
Done